### PR TITLE
Add support for automatic-restart for CloudSdkAppEngineDevServer1

### LINKED ISSUE #434: Add support for automatic-restart on file change for CloudSdkAppEngineDevServer1
_devappserver1_ supports `--automatic-restart`-like behaviour by setting the [`appengine.fullscan.seconds` system property](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/2125#issuecomment-315153081).

### PR DESCRIPTION
**Pull request by briandealwis**
Fixes #434

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java [hunk 1]
```diff
@@ -73,6 +73,9 @@ public void run(RunConfiguration config) throws AppEngineException {
     List<String> jvmArguments = new ArrayList<>();
     arguments.addAll(DevAppServerArgs.get("address", config.getHost()));
     arguments.addAll(DevAppServerArgs.get("port", config.getPort()));
+    if (Boolean.TRUE.equals(config.getAutomaticRestart())) {
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java]: I think we want to keep parameters that don't translate to dev_appserver2 out of this.

**Comment by chanseokoh** [comment]
Questions:

Have you tested "-Dappengine.fullscan.seconds=1"? Does it do an actual restart? Does it restart on changes to all of Java class files, JSPs, and static files?

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java (hunk 1 above)
Comment (briandealwis) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java]: This patch doesn't introduce `getAutomaticRestart()`: it's already there to support `dev_appserver2.py`'s `--automatic_restart=true`.

**Comment by briandealwis** [comment]
I've tested that `appengine.fullscan.seconds` reloads on .java files.  Will upload a server log snippet to illustrate what's going on once I've finished debugging a new issue.

Oddly, [auto-restart wasn't working for me on `dev_appserver2`](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/2125#issuecomment-315250771), though I haven't dug into it.

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java (hunk 1 above)
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java]: My fault, sorry, this should be fine.

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java [hunk 2]
```diff
@@ -73,6 +73,9 @@ public void run(RunConfiguration config) throws AppEngineException {
...
+      jvmArguments.add("-Dappengine.fullscan.seconds=1");
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java]: I think that 5 seconds is the value that is suggested by our docs though.

**Comment by chanseokoh** [comment]
> I've tested that `appengine.fullscan.seconds` reloads on .java files.

I'd like to check if the server actually restarts, not just reloading. Otherwise, I see the parameter name "automatic-restart" might confuse people later, wondering why it does not restart.

**Comment by briandealwis** [comment]
This parameter already exists and is modelled on what's provided by `dev_appserver.py`:
```sh
$ dev_appserver.py --help
[…]
  --automatic_restart [AUTOMATIC_RESTART]
                        restart instances automatically when files relevant to
                        their module are changed (default: True)
[…]
```

**Comment by chanseokoh** [comment]
Ah, I see it's about restarting instances, not the server. We're good.

**Comment by briandealwis** [comment]
devappserver1 only restarts on .class file changes, not .html changes

```
2017-07-14 13:57:54.439:INFO::main: Logging initialized @811ms
2017-07-14 13:57:55.002:INFO:oejs.Server:main: jetty-9.3.18.v20170406
2017-07-14 13:57:55.233:INFO:oeja.AnnotationConfiguration:main: Scanning elapsed time=24ms
2017-07-14 13:57:55.529:INFO:oejsh.ContextHandler:main: Started c.g.a.t.d.j.DevAppEngineWebAppContext@38467116{/,file:///Users/bsd/Manumitting/Projects/GooglePluginEclipse/runtime-gcloud-CLEAR/.metadata/.plugins/org.eclipse.wst.server.core/tmp0/aes/,AVAILABLE}{/Users/bsd/Manumitting/Projects/GooglePluginEclipse/runtime-gcloud-CLEAR/.metadata/.plugins/org.eclipse.wst.server.core/tmp0/aes}
2017-07-14 13:57:55.542:INFO:oejs.AbstractConnector:main: Started NetworkTrafficSelectChannelConnector@3feb2dda{HTTP/1.1,[http/1.1]}{localhost:8080}
2017-07-14 13:57:55.559:INFO:oejs.Server:main: Started @1932ms
Jul 14, 2017 5:57:55 PM com.google.appengine.tools.development.jetty9.JettyContainerService startHotDeployScanner
INFO: Full scan of the web app in place every 1s.
[…]
Jul 14, 2017 1:57:55 PM com.google.appengine.tools.development.DevAppServerImpl doStart
INFO: Dev App Server is now running
```
Change .java file, and republish:
```
Jul 14, 2017 2:00:50 PM com.google.appengine.tools.development.jetty9.JettyContainerService$2 filesChanged
INFO: A file has changed, reloading the web application.
2017-07-14 14:00:50.758:INFO:oejsh.ContextHandler:Scanner-0: Stopped c.g.a.t.d.j.DevAppEngineWebAppContext@38467116{/,file:///Users/bsd/Manumitting/Projects/GooglePluginEclipse/runtime-gcloud-CLEAR/.metadata/.plugins/org.eclipse.wst.server.core/tmp0/aes/,UNAVAILABLE}{/Users/bsd/Manumitting/Projects/GooglePluginEclipse/runtime-gcloud-CLEAR/.metadata/.plugins/org.eclipse.wst.server.core/tmp0/aes}
2017-07-14 14:00:50.764:INFO:oejs.AbstractConnector:Scanner-0: Stopped NetworkTrafficSelectChannelConnector@3feb2dda{HTTP/1.1,[http/1.1]}{localhost:8080}
2017-07-14 14:00:50.780:INFO:oejs.Server:Scanner-0: jetty-9.3.18.v20170406
2017-07-14 14:00:50.804:INFO:oeja.AnnotationConfiguration:Scanner-0: Scanning elapsed time=1ms
2017-07-14 14:00:50.830:INFO:oejsh.ContextHandler:Scanner-0: Started c.g.a.t.d.j.DevAppEngineWebAppContext@3faf459b{/,file:///Users/bsd/Manumitting/Projects/GooglePluginEclipse/runtime-gcloud-CLEAR/.metadata/.plugins/org.eclipse.wst.server.core/tmp0/aes/,AVAILABLE}{/Users/bsd/Manumitting/Projects/GooglePluginEclipse/runtime-gcloud-CLEAR/.metadata/.plugins/org.eclipse.wst.server.core/tmp0/aes}
2017-07-14 14:00:50.830:INFO:oejs.AbstractConnector:Scanner-0: Started NetworkTrafficSelectChannelConnector@3feb2dda{HTTP/1.1,[http/1.1]}{localhost:8080}
2017-07-14 14:00:50.831:INFO:oejs.Server:Scanner-0: Started @177205ms
```

**Pull request merged**